src/adapters/task/mapper/task.mapper.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable functional/no-let */
+import { TaskBuilder } from 'domain/task/builder/task/task.builder'
+import type { Task, UpdateTask } from 'domain/task/entity/task'
+import type { AmendTask } from 'domain/task/command/amendtask'
+import type { CreateTask } from 'domain/task/command/createTask'
+import type { DeepReadonly } from 'superTypes'
+import { UpdateTaskBuilder } from 'domain/task/builder/updateTask/updateTask.builder'
+
+export class TaskMapper {
+  public static readonly mapCreateTaskToEntity = (payload: DeepReadonly<CreateTask>): Task => {
+    let builder = new TaskBuilder()
+      .withId(payload.id)
+      .withCreationDate(payload.timestamp)
+      .withLastUpdateDate(payload.timestamp)
+      .withMessageKey(payload.messageKey)
+      .withType(payload.type)
+      .withStatus(payload.status)
+    if (payload.initiator) builder = builder.withInitiator(payload.initiator)
+    return builder.build()
+  }
+
+  public static readonly mapAmendTaskToEntity = (payload: DeepReadonly<AmendTask>): UpdateTask => {
```

Comment:
I hadn't noticed it before, but I find it strange that `UpdateTask` is an entity. There should be only one entity in the domain: `Task`.

Reply:
To be clearer:

**entities**:
- Task

**commands**:
- registerTask
- amendTask

events:
- TaskRegistered + TaskRegisteredPayload
- TaskAmended + TaskAmendedPayload